src/pages/en/guides/content-collections.mdx

```diff
@@ -79,7 +79,21 @@ To benefit from the full TypeScript and autocompletion features of [using schema
   </Fragment>
 </TypeScriptSettingTabs>
 
+### Set up the `.astro` directory
 
+Astro generates TypeScript types from your content collections in the `.astro` directory. These types will be updated anytime you run the [`astro dev`](/en/reference/cli-reference/#astro-dev), [`astro build`](/en/reference/cli-reference/#astro-build), or [`astro sync`](/en/reference/cli-reference/#astro-sync) commands.
+
+:::tip
+If you're using Git for version control, we recommend ignoring this generated directory by adding `.astro` to your `.gitignore`.
+:::
+
+These commands will also add a [reference path](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-path-) to your [`src/env.d.ts` file](/en/guides/typescript/#setup) if one exists. You can add this entry manually like so:
+
+```ts title="src/env.d.ts" ins={2}
+// use a relative path `../` to the `.astro` directory
+/// <reference types="../.astro/types.d.ts" />
+/// <reference path="astro/client" />
+````
```

Comment:
What do you think about being a bit more direct here: handled automatically, otherwise _do_ this. Because if you don’t you’re missing out right?

````suggestion
These commands will also add a [reference path](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-path-) to [`src/env.d.ts`](/en/guides/typescript/#setup) to make generated types available in your project. If your project does not include a `src/env.d.ts` file, create this file manually:

```ts title="src/env.d.ts"
// use a relative path `../` to the `.astro` directory
/// <reference types="../.astro/types.d.ts" />
/// <reference path="astro/client" />
````
````

Reply:
Seconding Chris' suggestion! (with the `ins={2}`) And, I would stat "If your project..." as its own paragraph. (personally)

Reply:
Updated with the separate paragraph. Are you sure about including `ins={2}`? My thinking was if you’re creating this manually, you need to create the _whole_ file, so highlighting just one line might be confusing?

Reply:
Well, looks like we're good to remove this! We now inject the whole `src/env.d.ts` file if none exists. Think we're safe to remove manual instructions 👍